src/object.d

```diff
@@ -4447,6 +4447,6 @@ unittest
     }
 
     int p;
-    scope arr = [S(&p)];
+    scope S[1] arr = [S(&p)];
```

Comment:
Isn't `scope` a bit useless if the data is on the stack?

Reply:
`scope` only applies to data on the stack.

Reply:
Would `S[1] arr = [S(&p)];` suffice?

Reply:
No. `S` needs to be stack allocated, not gc allocated. Hence the change.